lib/src/component_declaration/component_base.dart

```diff
@@ -841,6 +860,28 @@ abstract class _AccessorMetaCollection<T extends _Descriptor, U extends Accessor
     return meta ?? _emptyMeta;
   }
 
+  /// Returns a set of all the metadata in this collection
+  /// (for `propsMeta`, this corresponds to  all props mixins mixed into the props class).
+  Iterable<U> get all => _metaByMixin.values;
+
+  /// Returns a set of the metadata corresponding to [mixinTypes].
+  Iterable<U> forMixins(Set<Type> mixinTypes) =>
+      mixinTypes.map(forMixin);
+
+  /// Returns a set of all the metadata in this collection
+  /// (for `propsMeta`, this corresponds to  all props mixins mixed into the props class),
+  /// except for the metadata corresponding to [excludedMixinTypes].
+  Iterable<U> allExceptForMixins(Set<Type> excludedMixinTypes) {
+    final filtered = Map.of(_metaByMixin);
+    for (final mixinType in excludedMixinTypes) {
+      assert(_metaByMixin.containsKey(mixinType),
+      'No meta found for $mixinType;'
+          'it likely isn\'t mixed in by the props/state class.');
+      filtered.remove(mixinType);
+    }
+    return filtered.values;
+  }
+
```

Comment:
It might be helpful/useful to have a single point of documentation (possibly on the `consumedProps` getter) that enumerates all of the utilities that can/should be used.  Otherwise, these might not be all that discoverable / easily related.

* `PropsMeta.all`
* `PropsMeta.forSimpleKey`
* `PropsMeta.forMixin`
* `PropsMeta.forMixins`
* `PropsMeta.allExceptForMixins`

Reply:
Great idea!

Reply:
Hehe whoops say hello to my alter ego

![](https://i.kym-cdn.com/entries/icons/original/000/023/397/C-658VsXoAo3ovC.jpg)

Reply:
Addressed in 7d4248232f2ee6634df1584f147c293204ae8c97